R/games.R

```diff
@@ -909,13 +912,30 @@ random.graph.game <- erdos.renyi.game
 #' all(degree(powerlaw_vl_graph) == powerlaw_degrees)
 #'
 sample_degseq <- function(out.deg, in.deg = NULL,
-                          method = c("simple", "vl", "simple.no.multiple", "simple.no.multiple.uniform")) {
+                          method = c("configuration", "vl", "fast.heur.simple", "configuration.simple", "edge.switching.simple")) {
   method <- igraph.match.arg(method)
+
+  if (method == "simple") {
+    lifecycle::deprecate_warn("2.0.4", "sample_degseq(method = 'must be configuration instead of simple')")
+    method <- "configuration"
+  }
+
+  if (method == "simple.no.multiple") {
+    lifecycle::deprecate_warn("2.0.4", "sample_degseq(method = 'must be fast.heur.simple instead of simple.no.multiple')")
+    method <- "fast.heur.simple"
+  }
+
+  if (method == "simple.no.multiple.uniform") {
+    lifecycle::deprecate_warn("2.0.4", "sample_degseq(method = 'must be configuration.simple instead of simple.no.multiple.uniform')")
+    method <- "configuration.simple"
+  }
+
   method1 <- switch(method,
-    "simple" = 0,
+    "configuration" = 0,
     "vl" = 1,
-    "simple.no.multiple" = 2,
-    "simple.no.multiple.uniform" = 3
+    "fast.heur.simple" = 2,
+    "configuration.simple" = 3,
+    "edge.switching.simple" = 4 #TODO: check number and see where it comes from
```

Comment:
ideally we should make a note of where the numbers come from, for the next time someone needs to touch this part of the file.